src/gui/FileBrowser.cpp

```diff
@@ -567,10 +567,16 @@ void FileBrowserTreeWidget::keyPressEvent(QKeyEvent * ke )
 	else if (vertical || horizontal || preview || insert) { stopPreview(); }
 
 	// Try to get the currently selected item as a FileItem
-	auto file = dynamic_cast<FileItem*>(currentItem());
-	// If it's null (folder, separator, etc.), there's nothing left for us to do
-	if (file == nullptr) { return; }
+	auto item = currentItem();
 
+	auto file = dynamic_cast<FileItem*>(item);
+
+	// If it's not a file and the item is being previewed, assume it's a directory and try to expand or shrink it
+	if (!file && preview)
+	{
+			item->setExpanded(!item->isExpanded());
+			return;
+	}
```

Comment:
Fixing this for you since we want to return when its not a file regardless:
```suggestion
	if (!file)
	{
			// If it's not a file and the item is being previewed, assume it's a directory and try to expand or shrink it
			if (preview) { item->setExpanded(!item->isExpanded()); }
			return;
	}
```